share/picongpu/tests/PusherScaling/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,286 @@
+        # load series
+        series = io.Series(self.path, io.Access.read_only)
+        indices = np.array(series.iterations)
+        i = series.iterations[indices[0]]
+
+        if len(series.iterations) == 1:
+            raise ValueError("There is just 1 iteration in the series"
+                             "make sure, there are at least two")
+
+        elif len(np.array(i.particles)) != 1:
+            raise ValueError("There is not only 1 particle in the series"
+                             "make sure, there is only least one")
+
+        else:
+            # read parameters of the simulation
+            params = {
+                "cell_depth": i.get_attribute("cell_depth"),
+                "cell_height": i.get_attribute("cell_height"),
+                "cell_width": i.get_attribute("cell_width"),
+                "dt": i.get_attribute("dt"),
+                "unit_time": i.get_attribute("unit_time"),
+                "unit_length": i.get_attribute("unit_length"),
+                "unit_charge": i.get_attribute("unit_charge"),
+                "unit_bfield": i.get_attribute("unit_bfield"),
+                "unit_speed": i.get_attribute("unit_speed"),
+                "unit_mass": i.get_attribute("unit_mass"),
+                }
+            self.series = series
+            self.params = params
+
+    def read_series(self):
+        """Reads the position and momentum of the particle in the x-y-plane
+        provided by the simulation, as well as the charge and mass of the
+        particle.
+
+        Parameters:
+        series: simulation data
+
+        Returns: (self.variable)
+        x/y_poss (np.array): Relative in-Cell positions of the particle in the
+                             series
+        x/y_offSet (np.array): All node-positions of the particle in the series
+        periode (int): openPMD periode of the simulation
+        charge (float): Charge of particle
+        mass (float): Mass of particle
+        x/y_momentum (float): All momenta of the particle in the series
+        R_c (float): Radius of gyration of the particle at timestep 0
+
+        """
+
+        num_iter = len(self.series.iterations)
+
+        # create arrays for saving the data
+
+        # in-cell position
+        x_poss = np.zeros(num_iter)
+        y_poss = np.zeros(num_iter)
+
+        # cells
+        x_offSet = np.zeros(num_iter)
+        y_offSet = np.zeros(num_iter)
+
+        # momentum
+        x_momentum = np.zeros(num_iter)
+        y_momentum = np.zeros(num_iter)
+
+        i = self.series.iterations[0]
+        # charge
+        self.charge = (i.particles["e"]["charge"]
+                       [io.Mesh_Record_Component.SCALAR])
+        # mass
+        self.mass = i.particles["e"]["mass"][io.Mesh_Record_Component.SCALAR]
+
+        # calculating the openPMD-periode for later use
+        indices = np.array(self.series.iterations)
+        periode = indices[1] - indices[0]  # openPMD-periode
+
+        for iteration in indices:
```

Comment:
Doesn't something like this work as well:
```suggestion
        for i in self.series.iterations:
```

Reply:
Sadly this is not possible. For this case `i` would be an integer not an iteration...

Reply:
Okay.